redisbp/monitored_client.go

```diff
@@ -94,6 +100,40 @@ func (f MonitoredCmdableFactory) Close() error {
 	return f.client.Close()
 }
 
+// MonitorPoolStats starts a goroutine that publishes stats for the underlying
+// Redis client pool every 10 seconds using metricsbp.M.
+//
+// The goroutine will be closed when metricsbp.M.Ctx() is Done().
+func (f MonitoredCmdableFactory) MonitorPoolStats(tags metricsbp.Tags) {
+	t := tags.AsStatsdTags()
+	hitsGauge := metricsbp.M.RuntimeGauge(f.name + "-hits").With(t...)
+	missesGauge := metricsbp.M.RuntimeGauge(f.name + "-misses").With(t...)
+	timeoutsGauge := metricsbp.M.RuntimeGauge(f.name + "-timeouts").With(t...)
+	totalConnectionsGauge := metricsbp.M.RuntimeGauge(f.name + "-connections.total").With(t...)
+	idleConnectionsGauge := metricsbp.M.RuntimeGauge(f.name + "-connections.idle").With(t...)
+	staleConnectionsGauge := metricsbp.M.RuntimeGauge(f.name + "-connections.stale").With(t...)
+	client := f.BuildClient(context.TODO())
+	interval := 10 * time.Second
```

Comment:
this should also use `metricsbp.SysStatsTickerInterval` instead of hardcoded 10s?